### PR TITLE
impl: broadcaster [ISSUE-66]

### LINKED ISSUE #66: [comm] impl broadcaster
**Subject**

**Proposal Detail**

**Reference**

### PR DESCRIPTION
**Pull request by hyolin97**
## Related Issue

resolve: #66 

## Description

- implements broadcast

## Checklist

Thank you for your contribution.
Before submitting this PR, please make sure:

- [x] Applying goimports
- [X] Test case
- [X] End of Work

### CODE REVIEW

File: conn_internal_test.go [hunk 1]
```diff
@@ -0,0 +1,66 @@
...
+	connection := &MockStreamWrapper{}
+	con, _ := NewConnection(Address{"127.0.0.1", 8080}, "asdg", connection)
+	p.Add("asdf", con)
+	for r, _ := range p.connMap {
+		if r != "asdf" {
+			t.Fatalf("not added!")
+		}
+
+	}
+	//if(len(p.connMap)!=1){
```
Comment (AgwaB) [inline conn_internal_test.go]: remove this

File: conn_internal_test.go [hunk 2]
```diff
@@ -0,0 +1,66 @@
...
+
+func (c *MockStreamWrapper) GetStream() Stream {
+	return nil
+}
+
+func TestConnectionPool_Add(t *testing.T) {
+	p := NewConnectionPool()
+	connection := &MockStreamWrapper{}
+	con, _ := NewConnection(Address{"127.0.0.1", 8080}, "asdg", connection)
+	p.Add("asdf", con)
```
Comment (AgwaB) [inline conn_internal_test.go]: 1. `con` -> `conn`
2. Specify naming. not like `asdg`, `asdf`.
3. Add more test case

File: conn_internal_test.go [hunk 3]
```diff
@@ -0,0 +1,66 @@
...
+	for r, _ := range p.connMap {
+		if r != "asdf" {
+			t.Fatalf("not added!")
```
Comment (AgwaB) [inline conn_internal_test.go]: Specify error message. in this case, modify `Fatalf` to `Fatal`.

File: conn_internal_test.go [hunk 4]
```diff
@@ -0,0 +1,66 @@
...
+	//	t.Fatalf("Not added!")
+	//}
+}
+func TestConnectionPool_Remove(t *testing.T) {
+	p := NewConnectionPool()
+	connection := &MockStreamWrapper{}
+	con, _ := NewConnection(Address{"127.0.0.1", 8080}, "asdg", connection)
+	p.Add("asdf", con)
+	p.Remove("asdf")
+	if len(p.connMap) != 0 {
```
Comment (AgwaB) [inline conn_internal_test.go]: It'll be good to check test case' contents

File: conn_internal_test.go [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+		t.Fatalf("not deleted!")
+	}
+
+}
+func TestConnectionPool_Broadcast(t *testing.T) {
+	p := NewConnectionPool()
```
Comment (AgwaB) [inline conn_internal_test.go:62]: Plz complete the test or remove this

File: conn_internal_test.go [hunk 6]
```diff
@@ -0,0 +1,66 @@
...
+
+	}
+	//if(len(p.connMap)!=1){
+	//	t.Fatalf("Not added!")
+	//}
+}
+func TestConnectionPool_Remove(t *testing.T) {
+	p := NewConnectionPool()
+	connection := &MockStreamWrapper{}
+	con, _ := NewConnection(Address{"127.0.0.1", 8080}, "asdg", connection)
```
Comment (hihiboss) [inline conn_internal_test.go]: Change the name `asdg` to more meaningful name.

**Review by hihiboss — CHANGES_REQUESTED** [review]
Plz fix the conflict in `go.mod` and `go.sum`.

**Comment by hihiboss** [comment]
@AgwaB @zeroFruit @boohyunsik 

In type `Connection`, it has `Id() ConnID` method.
So, I think `Add(id ConnID, conn Connection)` could be changed to `Add(conn Connection)`.

Then, if you want to use `id ConnID`, just use `conn.Id()`!

How about this?

File: conn_internal_test.go [hunk 7]
```diff
@@ -0,0 +1,66 @@
...
+	p.Add("asdf", con)
+	p.Remove("asdf")
+	if len(p.connMap) != 0 {
+		t.Fatalf("not deleted!")
```
Comment (boohyunsik) [inline conn_internal_test.go]: Please write error message in more detail.

**Pull request closed without merging**